tests/hypervolume_tests/test_hssp.py

```diff
@@ -59,3 +59,61 @@ def test_solve_hssp_duplicated_infinite_loss() -> None:
         rank_i_loss_vals=test_case, rank_i_indices=np.arange(4), subset_size=2, reference_point=r
     )
     assert (0 not in res) or (1 not in res)
+
+
+def _solve_hssp_and_check_cache(
+    study: optuna.Study,
+    pareto_sols: np.ndarray,
+    pareto_indices: np.ndarray,
+    subset_size: int,
+    ref_point: np.ndarray,
+) -> np.ndarray:
+    selected_indices = optuna._hypervolume._solve_hssp_with_cache(
+        study, pareto_sols, pareto_indices, subset_size=subset_size, reference_point=ref_point
+    )
+    hssp_cache = study._storage.get_study_system_attrs(study._study_id)["hssp_cache"]
+    assert np.allclose(hssp_cache["rank_i_loss_vals"], pareto_sols)
+    assert np.array_equal(hssp_cache["rank_i_indices"], pareto_indices)
+    assert hssp_cache["subset_size"] == subset_size
+    assert np.allclose(hssp_cache["reference_point"], ref_point)
+    return selected_indices
+
+
+@pytest.mark.parametrize(
+    "storage",
+    (optuna.storages.RDBStorage("sqlite:///:memory:"), optuna.storages.InMemoryStorage()),
+)
+def test_solve_hssp_with_cache(storage: optuna.storages.BaseStorage) -> None:
```

Comment:
At first glance, it seems that this test includes too many preparations and logics. How about pre-determining the arguments of `_solve_hssp_and_check_cache`?